gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1308 @@
+
+            if memory_friendly_ttda is True, target can also be:
+            5. topic-term-distribution-array
+
+            example: [[0.1, 0.1, 0.8], [...], ...]
+
+            [topic1, topic2, ...]
+            with topic being an array of probabilities:
+            [token1, token2, ...]
+
+            token probabilities in a single topic sum to one, therefore, all the words sum to len(ttda)
+
+        num_new_models : integer, optional
+            the model keeps track of how many models were used in this ensemble. Set higher if ttda contained topics
+            from more than one model. Default: None, which takes care of it automatically.
+
+            If target is a 2D-array of float values, it assumes 1.
+
+            If the ensemble has ``memory_friendly_ttda`` set to False, then it will always use the number of models in
+            the target parameter.
+
+        """
+        # If the model has never seen a ttda before, initialize.
+        # If it has, append.
+
+        # Be flexible. Can be a single element or a list of elements
+        # make sure it's a numpy array
+        if not isinstance(target, (np.ndarray, list)):
+            target = np.array([target])
+        else:
+            target = np.array(target)
+            assert len(target) > 0
+
+        if self.memory_friendly_ttda:
+            # for memory friendly models/ttdas, append the ttdas to itself
+
+            detected_num_models = 0
+            ttda = []
+
+            # 1. ttda array, because that's the only accepted input that contains numbers
+            if isinstance(target.dtype.type(), (np.number, float)):
+                ttda = target
+                detected_num_models = 1
+
+            # 2. list of ensemblelda objects
+            elif isinstance(target[0], type(self)):
+                ttda = np.concatenate([ensemble.ttda for ensemble in target], axis=0)
+                detected_num_models = sum([ensemble.num_models for ensemble in target])
+
+            # 3. list of gensim models
+            elif isinstance(target[0], basemodel.BaseTopicModel):
+                ttda = np.concatenate([model.get_topics() for model in target], axis=0)
+                detected_num_models = len(target)
+
+            # unknown
+            else:
+                raise ValueError(f"target is of unknown type or a list of unknown types: {type(target[0])}")
+
+            # new models were added, increase num_models
+            # if the user didn't provide a custon numer to use
+            if num_new_models is None:
+                self.num_models += detected_num_models
+            else:
+                self.num_models += num_new_models
+
+        else:  # memory unfriendly ensembles
+            ttda = []
+
+            # 1. ttda array
+            if isinstance(target.dtype.type(), (np.number, float)):
+                raise ValueError(
+                    'ttda arrays cannot be added to ensembles, for which memory_friendly_ttda=False, '
```

Comment:
It'd be good to see this in the documentation, so that people know the costs/benefits of keeping memory friendly/unfriendly models.

Reply:
It already mentions it in the docstring:

```py
            if memory_friendly_ttda is True, target can also be:
            5. topic-term-distribution-array
```